src/encoder.rs

```diff
@@ -1561,12 +1567,14 @@ impl<'a, W: Write> Write for StreamWriter<'a, W> {
         self.to_write -= written;
 
         if self.index == self.line_len {
+            let mut filtered = vec![0; self.curr_buf.len()];
```

Comment:
Can we reuse this line buffer like we do in the decoder? But ultimately this is only a performance concern, small FIXME note would be more than enough.